include/alpaka/acc/AccGenericSycl.hpp

```diff
@@ -81,16 +81,10 @@ namespace alpaka
             : WorkDivGenericSycl<TDim, TIdx>{threadElemExtent, work_item}
             , gb::IdxGbGenericSycl<TDim, TIdx>{work_item}
             , bt::IdxBtGenericSycl<TDim, TIdx>{work_item}
-            , AtomicHierarchy<AtomicGenericSycl, AtomicGenericSycl, AtomicGenericSycl>{}
-            , math::MathGenericSycl{}
             , BlockSharedMemDynGenericSycl{dyn_shared_acc}
             , BlockSharedMemStGenericSycl{st_shared_acc}
             , BlockSyncGenericSycl<TDim>{work_item}
-            , IntrinsicGenericSycl{}
-            , MemFenceGenericSycl{}
-#    ifdef ALPAKA_DISABLE_VENDOR_RNG
-            , rand::RandDefault{}
-#    else
+#    ifndef ALPAKA_DISABLE_VENDOR_RNG
             , rand::RandGenericSycl<TDim>{work_item}
 #    endif
```

Comment:
I understand you wanted to keep only the strictly necessary initialisations.
Just for this case where one of the two `#ifdef` branches requires initialisation, would it make sense to keep also the other ?

Reply:
It does not make sense for me. I would only keep the code that really runs something.

Reply:
So... I should go ahead and remove all the code that is there only to silence warnings ?

Reply:
Anyway, whatever.

Reply:
> So... I should go ahead and remove all the code that is there only to silence warnings ?

I also dislike that sort of code. I am saddened that we have to add it to workaround compiler warnings.

But I am always happy if we can agree on dropping older compilers to remove some of that baggage!

Reply:
Discussed during today's alpaka meeting: no objections.